oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/service/ServiceTraffic.java

```diff
@@ -35,40 +35,63 @@
 import org.apache.skywalking.oap.server.core.storage.StorageHashMapBuilder;
 import org.apache.skywalking.oap.server.core.storage.annotation.Column;
 
+import static org.apache.logging.log4j.util.Base64Util.encode;
 import static org.apache.skywalking.oap.server.core.Const.DOUBLE_COLONS_SPLIT;
 
 @Stream(name = ServiceTraffic.INDEX_NAME, scopeId = DefaultScopeDefine.SERVICE,
     builder = ServiceTraffic.Builder.class, processor = MetricsStreamProcessor.class)
 @MetricsExtension(supportDownSampling = false, supportUpdate = false)
 @EqualsAndHashCode(of = {
     "name",
-    "nodeType"
+    "layer"
 })
 public class ServiceTraffic extends Metrics {
     public static final String INDEX_NAME = "service_traffic";
 
     public static final String NAME = "name";
-    public static final String NODE_TYPE = "node_type";
+
+    public static final String SHORT_NAME = "short_name";
+
+    public static final String SERVICE_ID = "service_id";
+
     public static final String GROUP = "service_group";
 
+    public static final String LAYER = "layer";
+
     @Setter
     @Getter
     @Column(columnName = NAME, matchQuery = true)
     private String name = Const.EMPTY_STRING;
 
     @Setter
     @Getter
-    @Column(columnName = NODE_TYPE)
-    private NodeType nodeType;
+    @Column(columnName = SHORT_NAME)
+    private String shortName = Const.EMPTY_STRING;
+
+    @Setter
+    @Column(columnName = SERVICE_ID)
+    private String serviceId;
 
     @Setter
     @Getter
     @Column(columnName = GROUP)
     private String group;
 
+    @Setter
+    @Getter
+    @Column(columnName = LAYER)
+    private Layer layer = Layer.undefined;
+
+    @Setter
+    private boolean isNormal = true;
+
     @Override
     protected String id0() {
```

Comment:
We need comments to explain what `id0` is about, and what is the `#serviceId` rule.

Reply:
Added